src/metabase/query_processor/middleware/wrap_value_literals.clj

```diff
@@ -51,10 +51,14 @@
                              (some #(when (= join-alias (:alias %))
                                       (:source-metadata %))
                                    (:joins inner-query))
-                             (:source-metadata inner-query))]
-      (some #(when (= (:name %) field-name)
-               (select-keys % [:base_type :effective_type :database_type]))
-            source-metadatas))))
+                             (:source-metadata inner-query))
+          [{[_ field-name :as field-ref] :field_ref :as field-metadata}] (filter #(= (:name %) field-name)
```

Comment:
Use `m/find-first` rather than `filter` + destructuring, if you only want the first match.